git-discover/src/upwards.rs

```diff
@@ -24,18 +26,21 @@ pub enum Error {
 
 /// Options to help guide the [discovery][function::discover()] of repositories, along with their options
 /// when instantiated.
-pub struct Options {
+pub struct Options<'a> {
     /// When discovering a repository, assure it has at least this trust level or ignore it otherwise.
     ///
     /// This defaults to [`Reduced`][git_sec::Trust::Reduced] as our default settings are geared towards avoiding abuse.
     /// Set it to `Full` to only see repositories that [are owned by the current user][git_sec::Trust::from_path_ownership()].
     pub required_trust: git_sec::Trust,
+    /// When discovering a repository, ignore any repositories that are located in these directories or any of their parents.
+    pub ceiling_dirs: &'a [PathBuf],
```

Comment:
The reason I also went with `Cow` for the list is that, a `Options::from_env` would need to use a vector to parse this.

Reply:
To me that seems alright, probably I am missing something. If you plan to submit a PR to integrate ceilings with `git-repository` we can tackle issues then.